netrd/distance/graph_diffusion_distance.py

```diff
@@ -0,0 +1,156 @@
+        :math:`e^{-t\mathcal{L}}` and :math:`e^{-t\mathcal{L}'}`, by
+        effectively simulating a diffusion process for :math:`t` timesteps,
+        creating a column vector of node-level activity at each timestep. The
+        distance :math:`d_\texttt{GDD}(G, G')` is defined as the Frobenius norm
+        between the two diffusion kernels at the timestep :math:`t^{*}` where
+        the two kernels are maximally different.
+
+        .. math::
+            D_{GDD}(G,G') = \sqrt{||e^{-t^{*}\mathcal{L}}-e^{-t^{*}\mathcal{L}'}||}
+
+        The results dictionary also stores a 2-tuple of the underlying
+        adjacency matrices in `adjacency_matrices`, the Laplacian matrices in
+        `laplacian matrices`, and the output of the optimization process
+        (`peak_diffusion_time` and `peak_deviation`).
+
+        Adapted from the authors' MATLAB code, available at: https://rb.gy/txbfrh
+
+
+        Parameters
+        ----------
+
+        G1, G2 (nx.Graph)
+            two networkx graphs to be compared.
+
+        thresh (float)
+            minimum value above which the eigenvalues will be considered.
+
+        resolution (int)
+            number of :math:`t` values to span through.
+
+        Returns
+        -------
+        dist (float)
+            the distance between `G1` and `G2`.
+
+        References
+        ----------
+
+        .. [1] Hammond, D. K., Gur, Y., & Johnson, C. R. (2013, December).
+               Graph diffusion distance: A difference measure for weighted graphs based on the
+               graph Laplacian exponential kernel. In Global Conference on Signal and
+               Information Processing, 2013 IEEE (pp 419-422). IEEE.
+               https://doi.org/10.1109/GlobalSIP.2013.6736904
+
+        """
+
+        A1 = nx.to_numpy_array(G1)
+        A2 = nx.to_numpy_array(G2)
+
+        L1 = laplacian(A1)
+        L2 = laplacian(A2)
+
+        D1, V1 = np.linalg.eig(L1)
+        D2, V2 = np.linalg.eig(L2)
+
+        eigs = np.hstack((np.diag(D1), np.diag(D2)))
+        eigs = eigs[np.where(eigs > thresh)]
+        eigs = np.sort(eigs)
+
+        if len(eigs) == 0:
+            dist = 0
+            self.results['dist'] = dist
+            return dist
+
+        t_upperbound = np.real(1.0 / eigs[0])
+        ts = np.linspace(0, t_upperbound, resolution)
+
+        # Find the Frobenius norms between all the diffusion kernels at
+        # different times. Return the value and where this vector is minimized.
+        E = -gdd_xi_t(V1, D1, V2, D2, ts)
+        f_val, t_star = (np.nanmin(E), np.argmin(E))
+
+        dist = np.sqrt(-f_val)
+
+        self.results['adjacency_matrices'] = A1, A2
+        self.results['laplacian_matrices'] = L1, L2
+        self.results['peak_diffusion_time'] = t_star
+        self.results['peak_deviation'] = f_val
+
+        self.results['dist'] = dist
+
+        return dist
+
+
+def gdd_xi_t(V1, D1, V2, D2, ts):
```

Comment:
Suggestion: I use `vals, vecs = np.eig()` always. For this I suggest `vals1, vecs2` and `vals2, vecs2`.

Reply:
Done. This is a common thing to do in the library, we should make sure our names are consistent (I'm sure they aren't.